dali/daliadmin/daliadmin.cpp

```diff
@@ -2794,114 +2798,177 @@ static void wuidCompress(const char *match, const char *type, bool compress)
     }
 }
 
-static void validateStore(bool fix, bool deleteFiles, bool verbose)
+static void validateStore(bool fix, bool deleteFiles, bool removeStaleLocations, bool verbose)
 {
     /*
      * Place holder for client-side dali store verification/validation. Currently performs:
      * 1) validates GeneratedDll entries correspond to current workunits (see HPCC-9146)
      */
     CTimeMon totalTime, ts;
 
-    PROGLOG("Gathering list of workunits");
-    Owned<IRemoteConnection> conn = querySDS().connect("/WorkUnits", myProcessSession(), RTM_LOCK_READ, 10000);
-    if (!conn)
-        throw MakeStringException(0, "Failed to connect to /WorkUnits");
+    std::unordered_set<std::string> machineMap;
     AtomRefTable wuids;
-    Owned<IPropertyTreeIterator> wuidIter = conn->queryRoot()->getElements("*");
-    ForEach(*wuidIter)
+
     {
-        IPropertyTree &wuid = wuidIter->query();
-        wuids.queryCreate(wuid.queryName());
-    }
-    PROGLOG("%d workunits gathered. Took %d ms", wuids.count(), ts.elapsed());
-    ts.reset(0);
+        PROGLOG("Gathering machines from environment");
+
+        ts.reset(0);
+        Owned<IRemoteConnection> envConn = querySDS().connect("/Environment", myProcessSession(), RTM_LOCK_READ, 10000);
+        if (!envConn)
+            throw makeStringException(0, "Failed to connect to /Environment");
 
-    StringArray uidsToDelete;
-    UnsignedArray indexToDelete;
+        Owned<IPropertyTreeIterator> machineIter = envConn->queryRoot()->getElements("Hardware/Computer");
+        ForEach(*machineIter)
+        {
+            const IPropertyTree &machine = machineIter->query();
+            const char *host = machine.queryProp("@netAddress");
+            IpAddress ip(host);
+            StringBuffer ipStr;
+            ip.getIpText(ipStr);
+            machineMap.insert(ipStr.str());
```

Comment:
should this also be inserting host - for when we store hostnames in the associated files?

Reply:
I think it's ok as it is, it can be a host, but is bound at this point.
When the dll locations are looked up, they could also be host names, and are bound before looking up.